sql/rpl_parallel.cc

```diff
@@ -2435,21 +2435,9 @@ rpl_parallel::stop_during_until()
 bool
 rpl_parallel::workers_idle()
 {
-  struct rpl_parallel_entry *e;
-  uint32 i, max_i;
-
-  max_i= domain_hash.records;
-  for (i= 0; i < max_i; ++i)
-  {
-    bool active;
-    e= (struct rpl_parallel_entry *)my_hash_element(&domain_hash, i);
-    mysql_mutex_lock(&e->LOCK_parallel_entry);
-    active= e->current_sub_id > e->last_committed_sub_id;
-    mysql_mutex_unlock(&e->LOCK_parallel_entry);
-    if (active)
-      break;
-  }
-  return (i == max_i);
+  DBUG_ASSERT(rli);
+  return rli->last_inuse_relaylog->queued_count ==
```

Comment:
For safety let's install an assert claiming equality for the two at slave start time, e.g into [rpl_parallel::reset()](https://github.com/MariaDB/server/blob/bb-10.4-MDEV-30619/sql/rpl_parallel.cc#L2275).